great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -762,6 +771,7 @@ def as_selectable(self) -> sqlalchemy.Selectable:
         raise NotImplementedError
 
 
+@public_api
```

Comment:
These assets are not dynamically defined, they just didn't have the `@public_api` decorator and so weren't showing up in our API docs.

But also I should note that users don't ever need to initiate these classes directly, we have factory methods that take care of that.